pyo3-macros-backend/src/pymethod.rs

```diff
@@ -160,11 +160,13 @@ enum PyMethodProtoKind {
 }
 
 impl<'a> PyMethod<'a> {
-    fn parse(
+    pub fn parse(
         sig: &'a mut syn::Signature,
         meth_attrs: &mut Vec<syn::Attribute>,
         options: PyFunctionOptions,
     ) -> Result<Self> {
+        ensure_function_options_valid(&options)?;
+        check_generic(sig)?;
```

Comment:
Is there a reason the order of these checks got switched from below? Otherwise I am 👍 for moving checks earlier in the macro phases.

Reply:
None, a mistake on my side. Fixed.